app/models/solidus_avatax_certified/line.rb

```diff
@@ -194,5 +184,45 @@ def shipment_cost(shipment)
       cost = shipment.discounted_amount.to_f
       cost.positive? ? order.taxable_shipping_total.to_f : 0
     end
+
+    def canada_ship_from
+      {
+        line1: '420 Green St',
+        line2: 'Unit 206',
+        city: 'Whitby',
+        region: 'ON',
+        country: 'CA',
+        postalCode: '19801'
+      }
+    end
+
+    def usa_ship_from
+      {
+        line1: '1000 N West St',
+        line2: 'STE 1200',
+        city: 'Wilmington',
+        region: 'DE',
+        country: 'US',
+        postalCode: '19801'
+      }
+    end
+
+    def ship_from_address(type, line_item = nil, shipment = nil)
+     return canada_ship_from if order.store.canada?
+
+     stock_loc = if line_item.present?
+                   line_item.inventory_units&.first&.shipment&.stock_location
+                 elsif shipment.present?
+                   shipment.stock_location
+                 end
+     return usa_ship_from unless stock_loc || stock_loc.state.abbr.in?(['MO','PA'])
+
+     case type
+     when 'item_line', 'return_line'
+       stock_loc.to_avatax_hash
+     when 'shipment_line'
+       shipment.stock_location.to_avatax_hash
```

Comment:
Is this `case...when..` really needed?
Won't this be same as `stock_loc.to_avatax_hash`?

Also - can we please use full name for variable please